src/contract_wrappers/exchange_wrapper.ts

```diff
@@ -183,6 +188,7 @@ export class ExchangeWrapper extends ContractWrapper {
             },
         );
         this._throwErrorLogsAsErrors(response.logs);
+        // this may throw and error if logs is empty
```

Comment:
We always emit at least one event from Fill.
Either LogFill or LogError.
Here we have an even stronger warranty, cause the line before this one throws on LogError.

Reply:
i think we should err on the side of caution and guard against an empty log here for a couple of reasons:
* its safer not to make assumptions about the implementation of the Exchange contract
* in the event that we do index an empty array, we want to give the user of the library a more descriptive error message